src/Tasks/Microsoft.Managed.After.targets

```diff
@@ -59,11 +57,22 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
   <ItemGroup Condition="'$(IsGraphBuild)' == 'true'">
     <ProjectReferenceTargets Include="Build" Targets="$(ProjectReferenceTargetsForBuildInOuterBuild)" Condition=" '$(ProjectReferenceTargetsForBuildInOuterBuild)' != '' " OuterBuild="true" />
+    <ProjectReferenceTargets Include="Build" Targets="GetTargetFrameworks" OuterBuild="true" SkipNonexistentTargets="true" />
     <ProjectReferenceTargets Include="Build" Targets="$(ProjectReferenceTargetsForBuild)" Condition=" '$(ProjectReferenceTargetsForBuild)' != '' " />
-
+    
     <ProjectReferenceTargets Include="Clean" Targets="$(ProjectReferenceTargetsForCleanInOuterBuild)" Condition=" '$(ProjectReferenceTargetsForCleanInOuterBuild)' != '' " OuterBuild="true" />
+    <ProjectReferenceTargets Include="Clean" Targets="GetTargetFrameworks" OuterBuild="true" SkipNonexistentTargets="true" />
     <ProjectReferenceTargets Include="Clean" Targets="$(ProjectReferenceTargetsForClean)" Condition=" '$(ProjectReferenceTargetsForClean)' != '' " />
 
+    <!--
+     Note: SkipNonexistentTargets="true" on the following two items means that an outer build node's call to its existent GetTargetFrameworks
+     target will fail if its inner build nodes don't define GetTargetFrameworksWithPlatformForSingleTargetFrameworks. This is necessary
+     since the P2P protocol cannot express logic that allows nonexistent targets to be skipped if other targets marked as SkipNonexistentTargets="true"
```

Comment:
I don't think this is the problem as much as there isn't a way to express the targets called from the outer build to the inner build.

What we have today effectively is saying "when `Build` is called on me, I call `GetTargetFrameworksWithPlatformForSingleTargetFramework` on my project references". But that isn't actually true at all. Instead what actually happens is that `GetTargetFrameworks` is called on the project references and then that calls `GetTargetFrameworksWithPlatformForSingleTargetFramework` on itself. We aren't able to express that here though, in particular "I call this target on my inner builds" so the middle part is cut out.